content/6_Advanced/LineContainer.mdx

```diff
@@ -30,13 +30,223 @@ frequency: 1
 	</Resource>
 </Resources>
 
-<IncompleteSection>Example problem + implementation?</IncompleteSection>
+<FocusProblem problem="HPIsam" />
 
-<Problems problems="half" />
+A half-plane can be defined as a planar region that consists of all points on one side of an infinite straight line, and none on the other side. We can apply computational geometry here, as an intersection can be represented as a convex polygon, where every point inside it is also inside all half-planes. If we can construct this polygon, we can retrieve the intersections.
 
-## LineContainer (aka $\mathcal{O}(N \log N)$ CHT)
+We can state that $N$ is the number of half-planes in our set. We represent lines and half-planes by a point and a vector, or in other terms, any point that lies on said line and the direction vector of the line. Each half-plane allows the region to the left of its direction vector. We can define the angle of a given half-plane as the polar coordinates of its direction vector. We can also state that a resulting intersection is always either bounded or empty, in cases where this doesn't hold, we can add 4 more half-planes which increase the size of the bounding box. For now, we shall also assume that parallel half-planes will not be present. To get a more visual representation of this, see the image below:
```

Comment:
> We can also state that a resulting intersection is always either bounded or empty

uuuhh what

Reply:
maybe in complex theory, but definitely not practically.

Reply:
it takes like 0 effort to come up with an infinite intersection

Reply:
ik but...

Reply:
doesn't show up in cp?